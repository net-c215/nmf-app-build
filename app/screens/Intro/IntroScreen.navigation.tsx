import React from "react";
import { Text } from "../../components";
import { t } from "../../utils";
import { Layout, ComponentsStyle } from "../../style";

const navigationOptions = () => ({
  headerStyle: {
    ...ComponentsStyle.header,
  },
  headerBackTitle: null,
  headerTitle: () => (
    <Text.H1 style={Layout.androidNavTitle}>
      {t("INTRO_SCREEN_WELCOME")}
    </Text.H1>
  ),
});

export default navigationOptions;
