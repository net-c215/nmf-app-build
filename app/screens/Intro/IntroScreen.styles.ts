import { StyleSheet } from "react-native";
import { PADDING_HORIZONTAL } from "../../constants/Layout";
import { Layout } from "../../style";

export default StyleSheet.create({
  welcomeView: {
    // backgroundColor: "green",
    flex: 1,
    justifyContent: "flex-end"
  },
  container: {
    ...Layout.containerWithPadding,
    justifyContent: "space-between"
  },
  image: {
    width: "60%"
  },
  imageView: {
    // backgroundColor: "orange",
    justifyContent: "center",
    flex: 2,
    alignItems: "center"
  },
  termsOfUseView: {
    // backgroundColor: "red",
    alignItems: "center",
    flex: 2
  },
  paragraph: {
    textAlign: "center",
    paddingVertical: 10
  },
  button: {
    position: "absolute",
    bottom: PADDING_HORIZONTAL,
    right: 0,
    left: 0
  }
});
