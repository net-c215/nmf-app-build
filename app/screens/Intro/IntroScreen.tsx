import React from "react";
import { SafeAreaView, View, Image, TouchableOpacity } from "react-native";
import * as WebBrowser from "expo-web-browser";
import { useSelector, useDispatch } from "react-redux";

import { Text, Button } from "../../components";
import ImagesAssets from "../../constants/ImagesAssets";
import styles from "./IntroScreen.styles";
import { t } from "../../utils";
import { userPreferences } from "../../ducks";

const currentTermsOfUseVersion = 1;

const IntroScreen = ({ navigation }) => {
  const dispatch = useDispatch();
  const acceptedTermsOfUseVersion = useSelector(
    userPreferences.selectors.getAcceptedTermsOfUseVersion
  );
  console.log(
    "TCL: IntroScreen -> acceptedTermsOfUseVersion",
    acceptedTermsOfUseVersion
  );
  if (currentTermsOfUseVersion === acceptedTermsOfUseVersion) {
    navigation.navigate("BudgetStack");
  }

  const onPress = () => {
    dispatch(
      userPreferences.actions.acceptTermsOfUse(currentTermsOfUseVersion)
    );

    navigation.navigate("BudgetStack");
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.welcomeView}>
        <Text.Primary bold style={styles.paragraph}>
          {t("INTRO_SCREEN_WELCOME")}
        </Text.Primary>
      </View>
      <View style={styles.imageView}>
        <Image
          style={styles.image}
          resizeMode="contain"
          source={ImagesAssets.stickers.restaurant}
        />
      </View>
      <View style={styles.termsOfUseView}>
        <Text.Secondary style={styles.paragraph}>
          {t("INTRO_SCREEN_PRELUDE")}
        </Text.Secondary>
        <Text.Tertiary style={styles.paragraph}>
          {t("INTRO_SCREEN_CONTINUE_AND_ACCEPT_TERMS_OF_USE")}
        </Text.Tertiary>
        <TouchableOpacity
          onPress={() =>
            WebBrowser.openBrowserAsync("http://nmf.earth/terms-of-use.pdf")
          }
        >
          <Text.Tertiary green>{t("INTRO_SCREEN_TERMS_OF_USE")}</Text.Tertiary>
        </TouchableOpacity>
        <Button.Primary
          fullWidth
          style={styles.button}
          onPress={onPress}
          textType={"Primary"}
        >
          <Text.Primary white center bold>
            {t("INTRO_SCREEN_I_AGREE")}
          </Text.Primary>
        </Button.Primary>
      </View>
    </SafeAreaView>
  );
};

export default IntroScreen;
