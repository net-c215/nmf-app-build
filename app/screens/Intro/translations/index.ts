import en from "./en";
import fr from "./fr";
import de from "./de";

interface TranslationKeys {
  INTRO_SCREEN_WELCOME: string;
  INTRO_SCREEN_PRELUDE: string;
  INTRO_SCREEN_CONTINUE_AND_ACCEPT_TERMS_OF_USE: string;
  INTRO_SCREEN_TERMS_OF_USE: string;
  INTRO_SCREEN_I_AGREE: string;
}

export { en, fr, de, TranslationKeys };
