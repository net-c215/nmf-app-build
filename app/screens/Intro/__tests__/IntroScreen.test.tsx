import React from "react";
import renderer from "react-test-renderer";
import Intro from "..";

const props = {
  navigation: {
    navigate: jest.fn()
  }
};

test("renders correctly ComingSoon", () => {
  const tree = renderer.create(<Intro {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
