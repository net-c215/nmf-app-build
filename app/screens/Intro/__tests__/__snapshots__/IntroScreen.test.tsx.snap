// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ComingSoon 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
      "paddingHorizontal": 14,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <Text.Primary
      bold={true}
      style={
        Object {
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Welcome to NMF.earth
    </Text.Primary>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "width": "60%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
      }
    }
  >
    <Text.Secondary
      style={
        Object {
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Are you ready to reduce your waste and transition to a sustainable life?
    </Text.Secondary>
    <Text.Tertiary
      style={
        Object {
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      By proceeding, you are agreeing to our terms of use.
    </Text.Tertiary>
    <TouchableOpacity
      onPress={[Function]}
    >
      <Text.Tertiary
        green={true}
      >
        Read terms of use
      </Text.Tertiary>
    </TouchableOpacity>
    <Button.Primary
      fullWidth={true}
      onPress={[Function]}
      style={
        Object {
          "bottom": 14,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
      textType="Primary"
    >
      <Text.Primary
        bold={true}
        center={true}
        white={true}
      >
        I agree
      </Text.Primary>
    </Button.Primary>
  </View>
</RCTSafeAreaView>
`;
